packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1189,13 +1190,33 @@ class WidgetsApp extends StatefulWidget {
 
   /// If true, forces the widget inspector to be visible.
   ///
+  /// Overrides the `debugShowWidgetInspector` value set in [WidgetsApp].
+  ///
+  /// Used by the `debugShowWidgetInspector` debugging extension.
+  ///
+  /// The inspector allows the selection of a location on your device or emulator
+  /// and view what widgets and render objects associated with it. An outline of
+  /// the selected widget and some summary information is shown on device and
+  /// more detailed information is shown in the IDE or DevTools.
+  static bool get debugShowWidgetInspectorOverride {
+    return debugShowWidgetInspectorOverrideNotifier.value;
+  }
+  static set debugShowWidgetInspectorOverride(bool value) {
+    debugShowWidgetInspectorOverrideNotifier.value = value;
+  }
+
+  /// If true, forces the widget inspector to be visible.
+  ///
+  /// Overrides the `debugShowWidgetInspector` value set in [WidgetsApp].
+  ///
   /// Used by the `debugShowWidgetInspector` debugging extension.
   ///
-  /// The inspector allows you to select a location on your device or emulator
+  /// The inspector allows the selection of a location on your device or emulator
   /// and view what widgets and render objects associated with it. An outline of
   /// the selected widget and some summary information is shown on device and
   /// more detailed information is shown in the IDE or DevTools.
-  static bool debugShowWidgetInspectorOverride = false;
+  @visibleForTesting
```

Comment:
We try to avoid this because at the end of the day it is still public API.

Instead of using `WidgetsApp.debugShowWidgetInspectorOverrideNotifier.value` the test could be using the public `WidgetsApp.debugShowWidgetInspectorOverride` right? That just leaves us with `WidgetsApp.debugShowWidgetInspectorOverrideNotifier.addListener`. The old version was testing this indirectly via `service.rebuildCount`. Any reason we can't continue to do that and keep the Notifier private? If we can't do that, we could at least lock down the public type of the `debugShowWidgetInspectorOverrideNotifier` to something like `Listenable` so it doesn't expose more API surface then necessary.

Reply:
I was able to overcome this by grabbing the Listenable from the ValueListenableBuilder, instead of exposing it in WidgetsApp :D Let me know what you think :D